classes/converter.php

```diff
@@ -96,6 +100,119 @@ final public function convert_moodle_url_to_pdf(moodle_url $url, string $filenam
         }
     }
 
+    /**
+     * Convert the given moodle URLs to PDF and store them in the file system.
+     * Note: If the currently logged in user does not have the correct capabilities to view the
+     * target URL, the created PDF will most likely be an error page.
+     *
+     * @param array $urls An array of target URLs to convert.
+     * @param string $filename the name to give converted file.
+     * (if none is specified, filename will be generated {@see \tool_pdfpages\helper::get_moodle_url_pdf_filename})
+     * @param array $options any additional options to pass to converter, valid options vary with converter
+     * instance, see relevant converter for further details.
+     * @param bool $keepsession should session be maintained after conversion? (For security reasons, this should always be `false`
+     * when conducting a conversion outside of a browser window, such as in an adhoc task or other background process, to prevent
+     * session hijacking.)
+     * @param string $cookiename cookie name to apply to conversion (optional).
+     * @param string $cookievalue cookie value to apply to conversion (optional).
+     * @param array $windowsize Size of the browser window. ex: `[1920, 1080]` (optional).
+     * @param string $useragent A custom User Agent to use when navigating the page (optional).
+     * @param string $outputfilepath The file path to store the output file.
+     * @param bool $printpagenumbers Whether to print page numbers in the footer of the combined PDF (optional,
+     *                               defaults to false).
+     *
+     * @return \stored_file the stored file created during conversion.
+     */
+    final public function convert_moodle_urls_to_pdf(array $urls, string $filename = '', array $options = [],
```

Comment:
Should we add some validation to ensure `$urls` is actually a list of `moodle_url`s ?

Reply:
Hi Cam, I pushed a fixup commit here https://github.com/catalyst/moodle-tool_pdfpages/pull/40/commits/17b313ae0e886ef0be9153ce2c62676724bc2547